docs/reference/github.md

```diff
@@ -0,0 +1,96 @@
+# GitHub Connector
+
+The [Elastic GitHub connector](../connectors/sources/github.py) is provided in the Elastic connectors python framework and can be used via [build a connector](https://www.elastic.co/guide/en/enterprise-search/current/build-connector.html).
+
+## Availability and prerequisites
+
+This connector is available as a **connector client** from the **Python connectors framework**. To use this connector, satisfy all [connector client requirements](https://www.elastic.co/guide/en/enterprise-search/master/build-connector.html).
+
+This connector is in **beta** and is subject to change. The design and code is less mature than official GA features and is being provided as-is with no warranties. Beta features are not subject to the support SLA of official GA features.
+
+## Usage
+
+To use this connector as a **connector client**, use the **GitHub** tile from the connectors list or **Customized connector** workflow.
+
+For additional operations, see [Usage](https://www.elastic.co/guide/en/enterprise-search/master/connectors-usage.html).
+
+## GitHub  access token
+
+Configure a GitHub  access token to fetch data from GitHub. 
+
+Follow these steps to generate a GitHub  access token:
+ - Go to GitHub Settings → Developer settings → Personal access tokens → Tokens(classic).
+ - Click `Generate new token`.
+ - Add a note and select the "repo" scope.
+ - Click `Generate token` and copy the token.
+
+## Configuration
+
+### Configure GitHub connector
+
+The following configuration fields are required:
+
+#### `github_token`
+
+GitHub  access token to authenticate the GitHub instance.
+
+#### `repositories`
+
+Comma-separated list of repositories to fetch data from GitHub instance. If the value is `*` the connector will fetch data from all repositories present in the configured user's account. Default value is `*`. Examples:
+
+- `Repository1, Repository2`
+- `*`
+
+#### `retry_count`
+
+The number of retry attempts after failed request to GitHub. Default value is `3`.
+
+### Content Extraction
+
+
+See [Content extraction](https://www.elastic.co/guide/en/enterprise-search/current/connectors-content-extraction.html).
```

Comment:
This needs more nuance, right? Since we're only fetching markdown and restructured-text files?

Reply:
It's fine if you'd rather capture that in a follow-up, as this was a recent change.

Reply:
ah, that's going into 8.10, and this is going into 8.9. Ignore me